bin/init.ml

```diff
@@ -30,17 +30,14 @@ let atom_printer ppf a = Format.pp_print_string ppf (Dune_lang.Atom.to_string a)
 let component_name_parser s =
   (* TODO refactor to use Lib_name.Local.conv *)
   let err_msg () =
-    User_error.make
-      [ Pp.textf "invalid component name `%s'" s
-      ; Lib_name.Local.valid_format_doc
-      ]
+    User_error.make [ Pp.textf "invalid component name `%s'" s ]
```

Comment:
To be user friendly, shouldn't we still give some feedback here about what counts as a valid component name?

Reply:
FWIU, There is no validation logic for ```Lib_name.of_string_opt```, i.e. it just creates ```Some s``` always, so there is no error message to display. Perhaps once this is decided - i.e. what constitutes as a valid lib name - we can revisit this again. 

As a matter of fact since we are never going to get any error here, we can remove the rest of the ```libs``` error handling code. But I want the PR to focus on solving a specific bug. The refactoring can come in another PR.

Reply:
Oh.. well, this verification was added to address #3046 I haven't had time to check this change (or to properly review it) to see the behavior, but I'm wondering how it avoids introducing a regression here...

Reply:
So, as I feared, this does introduce a regression. If I run

```sh
dune build @dune-init
```

we can see that invalid module names are no longer being caught: 

```
Done: 3926/3928 (jobs: 1)File "test/blackbox-tests/test-cases/dune-init/run.t", line 1, characters 0-0:
         git (internal) (exit 1)
(cd _build/default && /usr/bin/git diff --no-index --color=always -u test/blackbox-tests/test-cases/dune-init/run.t test/blackbox-tests/test-cases/dune-init/run.t.corrected)
diff --git a/test/blackbox-tests/test-cases/dune-init/run.t b/test/blackbox-tests/test-cases/dune-init/run.t.corrected
index 1f92984b..c3271187 100644
--- a/test/blackbox-tests/test-cases/dune-init/run.t
+++ b/test/blackbox-tests/test-cases/dune-init/run.t.corrected
@@ -239,13 +239,7 @@ Comments in dune files are preserved
 Will not create components with invalid names
 
   $ dune init lib invalid-component-name ./_test_lib
-  dune: NAME argument: invalid component name `invalid-component-name'
-        Library names must be non-empty and composed only of the
-        following
-        characters: 'A'..'Z', 'a'..'z', '_' or '0'..'9'.
-  Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
-  Try `dune init --help' or `dune --help' for more information.
-  [1]
+  Success: initialized library component named invalid-component-name
   $ test -f ./_test_lib
   [1]
```

I think a proper fix for this will need to a bit more subtle. I should have time to take a look tomorrow evening, and can provide some pointers on what I think might work, if that would be helpful!

Reply:
Commit https://github.com/ocaml/dune/pull/3444/commits/831900b9f1c34d2dd5aea1e4616233c67bf4b8a8 addresses test failure now. 

I have also made changes to the error mesage to be more specific now.